cfme/cloud/keypairs.py

```diff
@@ -50,19 +49,42 @@ def wait_for_delete(self):
 
     def create(self, cancel=False):
         """Create new keypair"""
-        navigate_to(self, 'All')
-        cfg_btn('Add a new Key Pair')
+        navigate_to(self, 'Add')
         fill(keypair_form, {'name': self.name, 'provider': self.provider.name},
              action=keypair_form.save_button)
         if not cancel:
-            flash.assert_message_match('Creating Key Pair {}'.format(self.name))
+            flash.assert_no_errors()
         else:
             flash.assert_message_match('Add of new Key Pair was cancelled by the user')
 
+    def add_tag(self, tag, **kwargs):
+        """Tags the system by given tag"""
+        navigate_to(self, 'All')
+        keypair_tbl.select_row_by_cells({'Name': self.name})
+        mixins.add_tag(tag, **kwargs)
+
+    def remove_tag(self, tag):
+        """Removes the selected tag off the system"""
+        navigate_to(self, 'All')
+        keypair_tbl.select_row_by_cells({'Name': self.name})
+        mixins.remove_tag(tag)
+
 
 @navigator.register(KeyPair, 'All')
 class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Key Pairs')
+
+    def resetter(self, refresh=False):
+        if refresh:
```

Comment:
**REQUIRED**
This method should not take an arg. To get this behavior, you would pass 'use_resetter=[bool]' to navigate(), where the default is `True`.

`navigate_to(obj, 'All', use_resetter=False)`